### PR TITLE
Fix Database Client Configuration Settings

### PR DESCRIPTION
**Pull request by sureshc**
In preparation for using RDS Proxy, I incorrectly created new, separate configuration settings for database username and password in #48314. RDS Proxy requires a single AWS Secret that contains username and password fields.  

- Add `db_credential_[user]` configuration settings to prepare for transitioning to fix this mistake. These settings are not used by any code yet, and will be modified to use the `!StackSecret` directive in #50285 once Stack Secrets (#50507) is merged. **No risk** to deploy because the new configuration settings are not used by any code in this branch.
- Rename the AWS Secret `DatabaseSecret` to `DatabaseSecretAdmin`. **Low risk** to deploy because in non-production environments, the admin SQL login password will be changed and updated on the cluster. Inventory of which environments are configured to use the admin login:
  - staging - globals.yml specifies writer username/password and writer endpoint populated from a Chef attribute. The deprecated Environment-specific `staging/cdo/db_admin_password` Secret is populated with the correct admin password.
  - test - globals.yml specifies writer username/password and writer endpoint populated from a Chef attribute. The deprecated Environment-specific `test/cdo/db_admin_password` Secret is populated with the correct admin password.
  - levelbuilder - globals.yml specifies writer username/password and writer endpoint populated from a Chef attribute. The deprecated Environment-specific `test/cdo/db_admin_password` Secret is populated with the correct admin password.
- Change the admin SQL user's password, as a side effect of renaming the AWS Secret above
- Modify the Daemon EC2 Instance CloudFormation template Resource to depend on RDS Proxy so that we don't attempt to connect to the database via the proxy from scheduled processes on the daemon before the proxies have completed provisioning.
-  Populate the `CDO.db_cluster_id`, `CDO.db_writer_endpoint`, and `CDO.db_proxy_reporting_endpoint` configuration settings from stack-specific secrets.
- Declare the new `CDO` database configuration settings with placeholder values. We will switch to using the new values in #50763 
- Stop using Chef recipe logic to configure the web application ("Front End" / non-daemon) EC2 Instances to use the RDS Proxy endpoint when connecting to the database

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by cat5inthecradle — APPROVED** [review]
This looks safe to me.

File: config.yml.erb [hunk 1]
```diff
@@ -520,10 +520,15 @@ db_cluster_id: !Secret
 db_proxy_reporting_endpoint: !Secret
 db_writer_endpoint: !Secret
 
+# Database credentials (username & password) are stored in an AWS Secret in a JSON format required by RDS Proxy:
+# https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/rds-proxy-setup.html#rds-proxy-secrets-arns
+db_credential_admin: !Secret
```
Comment (Hamms) [inline config.yml.erb]: should this be 
```suggestion
db_admin_credential: !Secret
```
for parallelization with the other values here?

**Review by Hamms — COMMENTED** [review]
I'm not entirely sure I follow what's going on here; are the credentials contained in this new variable going to be distinct from our exising `_user` and `_password` variables, or is it just an alternate way of presenting that information? If the latter, I'd much prefer for this value to be dynamically constructed from the others.

**Comment by sureshc** [comment]
> I'm not entirely sure I follow what's going on here; are the credentials contained in this new variable going to be distinct from our exising `_user` and `_password` variables, or is it just an alternate way of presenting that information? If the latter, I'd much prefer for this value to be dynamically constructed from the others.

The reasoning for this is buried in a comment. [RDS Proxy requires that we pass it a Secret](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/rds-proxy-setup.html#rds-proxy-secrets-arns) whose value contains both `username` and `password` properties so that it nows how to authenticate database clients connecting to it and so it has the credentials to connect to the actual database it is proxying. We won’t be storing username and password separately in Secrets Manager and our `CDO.` settings will store the combined JSON credential.

File: config.yml.erb (hunk 1 above)
Comment (sureshc) [reply config.yml.erb]: I went back and forth on this. I’m currently leaning towards the hierarchical naming convention: `db_credential_[user type]`:

- `db_credential_admin`
- `db_credential_writer`
- `db_credential_reader`

I definitely prefer for all the database configuration settings to start with the prefix `db_` but open to suggestions on how to organize the other settings.

Your suggestion has me realizing, I should change the endpoint settings to be consistent:

- `db_endpoint_writer`
- `db_endpoint_reporting`

Or I should change the credentials to the format you suggested.

**Pull request merged**